safe/gui/tools/minimum_needs/needs_profile.py

```diff
@@ -334,3 +331,20 @@ def remove_profile(self, profile):
             os.path.join(
                 str(self.root_directory), 'minimum_needs', profile + '.json')
         )
+
+    def load_old_profile(self, locale_minimum_needs_dir):
+        """Load old minimum needs profile under ~/.qgis2/minimum_needs.
+
+        :param locale_minimum_needs_dir: User local minimum needs profile path.
+        :type locale_minimum_needs_dir: str
+        """
+        old_profile_path = os.path.join(
+            QgsApplication.qgisSettingsDirPath(), 'minimum_needs')
+
+        if os.path.exists(old_profile_path):
+            for filename in os.listdir(old_profile_path):
+                source_file = os.path.join(old_profile_path, filename)
+                destination_file = os.path.join(
+                    locale_minimum_needs_dir, filename)
+                if not os.path.exists(destination_file):
+                    copy(source_file, destination_file)
```

Comment:
ah.. that's what I want to make sure in my first comment. 
> Can you add a check if the destination exists, then we should remove the file and the folder?

I thought you were telling me that we should remove the destination.
a lot of misunderstanding here haha.